File: ios/App/App/CategoriesManagerPlugin.m
//
//  CategoriesManagerPlugin.m
//  App
//
//  Created by Dariusz Zabrzeński on 01/03/2023.
//

#import <Capacitor/Capacitor.h>

CAP_PLUGIN(CategoriesManagerPlugin, "CategoriesManagerPlugin",
           CAP_PLUGIN_METHOD(add, CAPPluginReturnNone);
           CAP_PLUGIN_METHOD(get, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(remove, CAPPluginReturnNone);
)

File: ios/App/App/CategoriesManagerPlugin.swift
//
//  CategoriesManagerPlugin.swift
//  App
//
//  Created by Dariusz Zabrzeński on 01/03/2023.
//

import Capacitor


@objc(CategoriesManagerPlugin)
public class CategoriesManagerPlugin: CAPPlugin {
    @objc func add(_ call: CAPPluginCall) {
        let cm = CategoriesManager()
        
        let text = call.getString("text")!
        cm.add(text: text)
        call.resolve()
    }
    
    @objc func get(_ call: CAPPluginCall) {
        let cm = CategoriesManager()
        let data = cm.get()
        var result:[Any] = []
        
        for item in data.items {
            let itemDict: [String: Any] = [
                "text": item.text,
                "id": item.id
            ]
            result.append(itemDict)
        }
        
        call.resolve(["items": result])
    }
    
    @objc func remove(_ call: CAPPluginCall) {
        let cm = CategoriesManager()
        
        let id = call.getInt("id")!
        cm.remove(id: id)
        call.resolve()
    }
}

